From: Thomas Hindoe Paaboel Andersen <phomes@gmail.com>
Date: Fri, 16 May 2014 23:46:48 +0200
Subject: udev: avoid use of uninitialized err

After 1ea972174baba40dbc80c51cbfc4edc49764b59b err is no longer
set unless we hit a special case. Initialize it to 0 and remove
a check that will never fail.

(cherry picked from commit bf9bead187802a52a1f376a03caee762d663e945)

Conflicts:
	src/udev/udevd.c
---
 src/udev/udevd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/udev/udevd.c b/src/udev/udevd.c
index 93afca11..73afa6f9 100644
--- a/src/udev/udevd.c
+++ b/src/udev/udevd.c
@@ -272,7 +272,7 @@ static void worker_new(struct event *event)
                 for (;;) {
                         struct udev_event *udev_event;
                         struct worker_message msg;
-                        int err;
+                        int err = 0;
 
                         log_debug("seq %llu running", udev_device_get_seqnum(dev));
                         udev_event = udev_event_new(dev);
@@ -293,7 +293,7 @@ static void worker_new(struct event *event)
                         udev_event_execute_run(udev_event, &sigmask_orig);
 
                         /* apply/restore inotify watch */
-                        if (err == 0 && udev_event->inotify_watch) {
+                        if (udev_event->inotify_watch) {
                                 udev_watch_begin(udev, dev);
                                 udev_device_update_db(dev);
                         }
